core/src/test/java/org/jdbi/v3/core/mapper/reflect/ConstructorMapperTest.java

```diff
@@ -298,4 +299,36 @@ public void nestedPrefixParametersStrict() {
             this.nested = nested;
         }
     }
+
+    @Test
+    public void testFactoryMethod() {
+        StaticFactoryMethodBean bean = selectOne("SELECT s, i FROM bean", StaticFactoryMethodBean.class);
+
+        assertThat(bean.s).isEqualTo("3");
+        assertThat(bean.i).isEqualTo(2);
+    }
+
+    @Test
+    public void testFactoryMethodReversed() {
+        StaticFactoryMethodBean bean = selectOne("SELECT i, s FROM bean", StaticFactoryMethodBean.class);
+
+        assertThat(bean.s).isEqualTo("3");
+        assertThat(bean.i).isEqualTo(2);
+    }
+
+    static class StaticFactoryMethodBean {
+        private final String s;
+        private final int i;
+
+        StaticFactoryMethodBean(String s, int i, int pass) {
+            assertThat(pass).isEqualTo(42); // Make sure this constructor is not used directly
+            this.s = s;
+            this.i = i;
+        }
+
+        @JdbiConstructor
+        static StaticFactoryMethodBean create(String s, int i) {
+            return new StaticFactoryMethodBean(s, i, 42);
+        }
+    }
```

Comment:
Can we a test for multiple static methods with the `@JdbiConstructor` annotation?